src/SpecValidatorInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Dhii\Validation;
+
+use Traversable;
+
+/**
+ * Something that can validate a value according to a specification.
+ *
+ * @since [*next-version*]
+ */
+interface SpecValidatorInterface extends ValidatorInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     *
+     * @param array|Traversable|null $spec A list of criteria, a.k.a. the spec, if any.
+     */
+    public function validate($value, $spec = null);
```

Comment:
Why not simply default `$spec` to `array()` and remove the null typehint?

Reply:
Trying to standardize params to have default `null` value as much as possible. Besides, if I make it an array, I won't be able to tell if no spec is passed, or the spec is simply empty.

Reply:
That's fair.